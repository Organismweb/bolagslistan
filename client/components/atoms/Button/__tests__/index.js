import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { Button, TomatoButton } from "../index";

test("normal button renders", () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("color", "grey");
});

test("tomato button renders", () => {
  const tree = renderer.create(<TomatoButton />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("color", "tomato");
  expect(tree).toHaveStyleRule("color", "red", {
    modifier: ":hover"
  });
});
