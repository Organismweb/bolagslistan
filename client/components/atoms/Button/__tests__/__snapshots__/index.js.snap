// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normal button renders 1`] = `
.c0 {
  display: inline-block;
  padding: 10px 20px;
  color: grey;
}

<button
  className="c0"
/>
`;

exports[`tomato button renders 1`] = `
.c0 {
  display: inline-block;
  padding: 10px 20px;
  color: grey;
  color: tomato;
}

<button
  className="c0"
/>
`;
